clients/src/test/java/org/apache/kafka/clients/consumer/internals/FetcherTest.java

```diff
@@ -637,14 +637,55 @@ public void testQuotaMetrics() throws Exception {
     }
 
     @Test
-    public void testGetOffsetsByTimes() {
-        client.prepareResponseFrom(listOffsetResponse(Errors.NONE, 100L, 100L), cluster.leaderFor(tp));
+    public void testGetOffsetsByTimesWithErrors() {
+        // Error code none with unknown offset
+        testGetOffsetsByTimesWithError(Errors.NONE, -1L, 100L, null, 100L);
+        // Error code none with known offset
+        testGetOffsetsByTimesWithError(Errors.NONE, 10L, 100L, 10L, 100L);
+        testGetOffsetsByTimesWithError(Errors.NOT_LEADER_FOR_PARTITION, 10L, 100L, 10L, 100L);
+        testGetOffsetsByTimesWithError(Errors.UNKNOWN_TOPIC_OR_PARTITION, 10L, 100L, 10L, 100L);
+        testGetOffsetsByTimesWithError(Errors.INVALID_REQUEST, 10L, 100L, null, 100L);
+        testGetOffsetsByTimesWithError(Errors.BROKER_NOT_AVAILABLE, 10L, 100L, 10L, 100L);
+    }
+
+    // Test the getOffsetsForTimes behavior when error occurs.
+    // The error is always on partition 0 on the first try.
```

Comment:
What if the second response has an error or both responses do? Are these cases worth covering too?